app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -633,6 +514,190 @@ public function handleTransactionUpdate(Mage_Payment_Model_Info $payment, $newTr
             throw $e;
         }
     }
+    
+    public function handleRefundTransactionUpdate(
+        Mage_Payment_Model_Info $payment,
+        $newTransactionStatus,
+        $prevTransactionStatus,
+        $transactionAmount = null,
+        $transaction
+    ) {
+        try {
+            $newTransactionStatus = strtolower($newTransactionStatus);
+            if ($prevTransactionStatus == null || $newTransactionStatus !== self::TRANSACTION_REFUND) {
+                return;
+            }
+            $prevTransactionStatus = strtolower($prevTransactionStatus);
+            $this->validateWebHook($newTransactionStatus, $prevTransactionStatus);
+
+            $order             = $payment->getOrder();
+            $transactionAmount = Mage::app()->getStore()->roundPrice($transactionAmount);
+            $totalRefunded     = $order->getTotalRefunded() ?: 0;
+            $totalPaid         = $order->getTotalPaid();
+            $availableRefund   = Mage::app()->getStore()->roundPrice(
+                $totalPaid - $totalRefunded
+            );
+            if ($availableRefund < $transactionAmount) {
+                $message = Mage::helper('boltpay')->__('Maximum amount available %s is less than requested %s',
+                    $availableRefund, $transactionAmount);
+                Mage::throwException($message);
+            }
+
+            $service         = Mage::getModel('sales/service_order', $order);
+            $invoiceIds      = $order->getInvoiceCollection()->getAllIds();
+            $isPartialRefund = false;
+            //actually for order with bolt payment, there is only one invoice can refund
+            if ($invoiceIds && isset($invoiceIds[0])) {
+                $refundTransactionId       = $transaction->id;
```

Comment:
In the future, could you add a comment here explaining how this solves the problem? It's fine now with context and with the asana task fresh in mind, but it's useful if someone needs to look at this PR in the future and understand the change without having reviewed it initially.

It also makes reviews quicker since this is the only part that significantly changed.